src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Vms/TotalTestResultEntryVmComparer.cs

```diff
@@ -7,16 +7,18 @@ public class TotalTestResultEntryVmComparer : IEqualityComparer<TotalTestResultE
     {
         public bool Equals(TotalTestResultEntryVm x, TotalTestResultEntryVm y)
         {
-#pragma warning disable CA1062 // Validate arguments of public methods
-            return x.UserId == y.UserId;
-#pragma warning restore CA1062 // Validate arguments of public methods
+            if (x != null && y != null)
+                {
```

Comment:
same as above

also, rename x and y variables to something more understandable